demos/demo/main.go

```diff
@@ -330,3 +319,23 @@ func newBarChart() *tvxwidgets.BarChart {
 
 	return barGraph
 }
+
+// Source: https://stackoverflow.com/questions/50833673/rotate-array-in-go/79079760#79079760
+// rotate rotates the given slice by k positions to the left or right.
+func rotate[T any](slice []T, k int) []T {
```

Comment:
I have added this here because the rotation of the sinus and cosin data was broken after some time, leading to strange effects in the demo graph scaling when set to automatic, as well as showing a "flat line" after a while. This method can be used in other places as well an works reliably. Although I have added a source, the source is me ;)